flex/engines/http_server/actor/executor.act.cc

```diff
@@ -50,4 +56,307 @@ seastar::future<query_result> executor::run_graph_db_query(
+  }
+  try {
+    result["values"] = gs::VertexEdgeManager::getVertex(
+        vertex_data, hiactor::local_shard_id());
+    return seastar::make_ready_future<admin_query_result>(
+        gs::Result<seastar::sstring>(result.dump()));
+  } catch (std::exception& e) {
+    return errorResponse(gs::StatusCode::InvalidSchema,
+                         "fail to get vertex : " + std::string(e.what()));
+  }
+}
+
+seastar::future<admin_query_result> executor::get_edge(
+    graph_management_query_param&& param) {
+  nlohmann::json schema_json, result;
+  std::vector<gs::VertexData> vertex_data;
+  std::vector<gs::EdgeData> edge_data;
+  // Parse the input json
+  auto query_params = std::move(param.content.second);
+  //  Check if the currently running graph is graph_id
+  try {
+    schema_json = getSchemaData(metadata_store_, param.content.first);
+  } catch (const std::exception& e) {
+    return errorResponse(gs::StatusCode::NotFound,
+                         "Graph not exists : " + std::string(e.what()));
+  }
+  // input edge data
+  try {
+    gs::EdgeData edge;
+    std::string src_label = query_params["src_label"];
+    std::string dst_label = query_params["dst_label"];
+    std::string edge_label = query_params["edge_label"];
+    std::string src_pk_value = query_params["src_primary_key_value"];
+    std::string dst_pk_value = query_params["dst_primary_key_value"];
+    edge.src_pk_value = gs::Any(src_pk_value);
+    edge.dst_pk_value = gs::Any(dst_pk_value);
+    gs::VertexEdgeManager::checkEdgeSchema(schema_json, edge, src_label,
+                                           dst_label, edge_label);
+    gs::VertexEdgeManager::getEdgeLabelId(
+        edge, src_label, dst_label, edge_label, hiactor::local_shard_id());
+    edge_data.push_back(edge);
+    result["src_label"] = src_label;
+    result["dst_label"] = dst_label;
+    result["edge_label"] = edge_label;
+    result["src_primary_key_value"] = src_pk_value;
+    result["dst_primary_key_value"] = dst_pk_value;
+  } catch (std::exception& e) {
+    return errorResponse(gs::StatusCode::InvalidSchema,
+                         " Bad input parameter : " + std::string(e.what()));
+  }
+  try {
+    result["properties"] =
+        gs::VertexEdgeManager::getEdge(edge_data, hiactor::local_shard_id());
+    return seastar::make_ready_future<admin_query_result>(
+        gs::Result<seastar::sstring>(result.dump()));
+  } catch (std::exception& e) {
+    return errorResponse(gs::StatusCode::InvalidSchema,
+                         "fail to get edge : " + std::string(e.what()));
+  }
+}
+
+seastar::future<admin_query_result> executor::delete_vertex(
+    graph_management_param&& param) {
+  return errorResponse(gs::StatusCode::NotFound,
+                       "delete_vertex is not implemented");
+}
+
+seastar::future<admin_query_result> executor::delete_edge(
+    graph_management_param&& param) {
+  return errorResponse(gs::StatusCode::NotFound,
```

Comment:
StatusCode::NotImplemented

Reply:
I think you mean StatusCode::Unimplemented, I've replaced it.